src/RuleFilterParser/Api/ApiClientExtension.cs

```diff
@@ -0,0 +1,19 @@
+﻿namespace RuleFilterParser;
+
+public static class ApiClientExtension
+{
+    // TODO this design leads to a lot of 'Provider' objects in memory. We should redesign this so that the provider is instantiated once as long as the API client lives.
+
+    /// <summary>
+    /// Creates a Queryable object
+    /// </summary>
+    /// <param name="apiClient"></param>
+    /// <param name="path"></param>
+    /// <typeparam name="T"></typeparam>
+    /// <returns></returns>
+    public static IQueryable<T> GetAsQueryable<T>(this IApiClient apiClient, string path)
```

Comment:
The example in Core.Api suits better as it creates a client as a transient scope.

```
            services.AddHttpClient();
            services.AddTransient<ICoreApiClient>(x =>
            {
                var clientFactory = x.GetRequiredService<IHttpClientFactory>();
                return new CoreApiClient(options, clientFactory);
            });
            services.AddTransient(x => x.GetRequiredService<ICoreApiClient>().Persons );
            services.AddTransient(x => x.GetRequiredService<ICoreApiClient>().Countries );
            services.AddTransient(x => x.GetRequiredService<ICoreApiClient>().Relations );
            services.AddTransient(x => x.GetRequiredService<ICoreApiClient>().Affiliations );
            services.AddTransient(x => x.GetRequiredService<ICoreApiClient>().Consents );
            services.AddTransient(x => x.GetRequiredService<ICoreApiClient>().Orgs );
```

Reply:
Yes. I see we need something like [`DbSet<T>`](https://learn.microsoft.com/en-us/dotnet/api/microsoft.entityframeworkcore.dbset-1?view=efcore-2.0) for the API...

Reply:
Please check out my new version of the sample [ApiClientMockup](https://github.com/leo-schick/rule-filter-parser-dotnet/blob/query-provider/tests/RuleFilterParser.Tests/Mockups/ApiClientMockup.cs#L6-L15) class.

I renamed now the extension method from `GetAsQueryable` to `GetQueryable` to have the same naming as the Core API. But the generic type parameter and the `path` parameter are still required.

I am not anymore sure if we should keep the extension method. It would be possible that the ApiClient holds a instance of `ApiQueryProvider` as well. The static typed properties could instanciate a `ApiQueryable` instance on calling. I did not spend so much time on how the queryable logic should be integrated with the API client... maybe someone has better ideas